src/VisualStudio/Core/Impl/CodeModel/InternalElements/CodeFunctionDeclareDecl.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeModel.InternalElements
+{
+    public sealed class CodeFunctionDeclareDecl : CodeFunction
+    {
+        internal static new EnvDTE.CodeFunction Create(
+          CodeModelState state,
+          FileCodeModel fileCodeModel,
+          SyntaxNodeKey nodeKey,
+          int? nodeKind)
+        {
+            var element = new CodeFunctionDeclareDecl(state, fileCodeModel, nodeKey, nodeKind);
+            var result = (EnvDTE.CodeFunction)ComAggregate.CreateAggregatedObject(element);
+
+            fileCodeModel.OnElementCreated(nodeKey, (EnvDTE.CodeElement)result);
+
+            return result;
+        }
+
+        internal static new EnvDTE.CodeFunction CreateUnknown(
```

Comment:
@Pilchie That was a good catch. I forgot to wire it. I have wired it properly and added testcases for that.